core/src/main/java/com/yahoo/oak/ScopedReadBuffer.java

```diff
@@ -25,12 +25,12 @@
         this.s = other.s.getDuplicatedSlice();
     }
 
-    protected int getDataOffset(int index) {
+    protected long getDataOffset(int index) {
```

Comment:
this doesnt return the address of the data it really returns the offset, which I think is never used since we can get the exact address of the data using the getAddress() method

Reply:
It returns the address and it is used. Read the rest of the file :)

Reply:
yep my mistake was thinking of something else